torch/_higher_order_ops/cond.py

```diff
@@ -28,16 +30,136 @@
 )
 
 
+@contextmanager
+def _set_compilation_env():
+    _old_is_tracing = torch.fx._symbolic_trace._is_fx_tracing_flag
+    _old_size_limit = torch._dynamo.config.cache_size_limit
+
+    try:
+        # We need to turn off the is_fx_tracing_flag. Remove this flag check from dyanmo
+        # once we are confident fx tracing works with dynamo.
+        torch.fx._symbolic_trace._is_fx_tracing_flag = False
```

Comment:
Sorry, I still don't understand why we are turning this flag to False. You mentioned that making it go to True would make a lot of tests fail. But we shouldn't be regressing behavior, unless they're related to the symbool thing?

Reply:
make_fx will 1. turn this flag on and 2. feed FakeTensors and SymBools into torch.compile. I kind of assume this flag can be turned off in cond_compiled and it's not a problem. 

Right now, if we leave this flag alone, we'll get an error for all make_fx tests, which isn't what we want we already supports "real" tracing mode.

Reply:
Is the problem that in torch.compile, we want `_is_fx_tracing_flag` to be False? If so, I would have expected that we turned that off inside of torch.compile, because this is a make_fx and torch.compile composability issue, rather than doing a one-off solution here